server/api/initial_data.py

```diff
@@ -571,14 +571,20 @@ def _migrate_artifacts_batch(
         # project - copy as is
         new_artifact.project = artifact_metadata.get("project", None)
 
-        # key - the artifact's key, without iteration if it is attached to it
-        key = artifact_metadata.get("key", "")
-        new_artifact.key = key
-
         # iteration - the artifact's iteration
         iteration = artifact_metadata.get("iter", None)
         new_artifact.iteration = int(iteration) if iteration else 0
 
+        # key - the artifact's key, without iteration if it is attached to it
+        key = (
```

Comment:
Does it mean that we will never be able to remove the db_key from the spec?
I don't really understand what was the issue. The query is built in the API and users only use `metadata.key` don't they? The backend knows that the key was changed so it should be able to make the adjustment.

Reply:
The problem is that the key isn't the Metadata can be different than the db_key.
When reading/listing artifacts, you can only query on the key that is saved in the DB -> `db_key`.

So if users used to query on the DB key, after the migration they wouldn't be able to.

Even in the new API, the BE doesn't auto-resolve the two options, it can only filter on the key column (the rest is kept in the full object).